test/ClientTest.cs

```diff
@@ -71,5 +73,26 @@ public void ActingAsTest()
 
             Assert.AreEqual(false, plainClientVars.MoveNext());
         }
+
+        [Test]
+        public void CreatePolicyTest()
+        {
+            string policyId = "vaultname/policyname";
+            string policyPath = $"test:///policies/{Client.GetAccountName()}/{Constants.KIND_POLICY}";
+
+            // notice: We must encode policyId, we can check that sensitive data is not leaked (123456)
+            Mocker.Mock(new Uri($"{policyPath}/{WebUtility.UrlEncode(policyId)}"), "apisecret:123456");
+
+            Client.Authenticator = new MockAuthenticator();
+
+            using (MemoryStream ms = new MemoryStream())
+            {
+                using (StreamWriter sw = new StreamWriter(ms, Encoding.Unicode))
+                {
+                    sw.WriteLine("-!policy id: database body: -!host id: db - host - !variable id: db - password owner: !host db - host");
+                    new Policy(Client, policyId).LoadPolicy(ms);
+                }
+            }
+        }
```

Comment:
What are we testing for here?

As policy loading is the crux of the vault synchronizer project, we really need to get some integration tests in place to validate the full end-to-end flow.  

Check out the `cyberark/conjur-api-go` project's `test.sh` file for reference: https://github.com/cyberark/conjur-api-go/blob/master/test.sh.  It uses docker compose to start a Conjur instance and test container, then runs all tests from the test container.

Reply:
You are absolutely right, we need to add Integration framework to this SDK, currently tested manually and ran the example program until we support integration

Reply:
Let's start with at least a couple valid unit tests. I'm really uncomfortable continuing to develop this library without tests. Tests are key to verifying functionality and insuring changes don't impact existing functionality. Manual testing provides no visibility (I can't tell if anyone's run this code to verify it works as expected). 

To start with, we could mock the request and response to make sure the code does what it is supposed to.  In addition, we'd really benefit from an integration test that:
1. Loads a policy with a variable
2. Sets the value of variable
3. Retrieves the variable and validates the value.
as this is the primary use case of the library.  

Please add some valid tests and we'll get this merged.